verification.go

```diff
@@ -0,0 +1,101 @@
+package neutrino
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcutil"
+	"github.com/btcsuite/btcutil/gcs"
+	"github.com/btcsuite/btcutil/gcs/builder"
+)
+
+// VerifyBasicBlockFilter asserts that a given block filter was constructed
```

Comment:
We can likely consolidate some of this logic with what we have in `resolveFilterMismatchFromBlock`. This is called when our peers advertise a differing set of filters and we want to verify as much as we can. As is, it only verifies that all the pkScripts produced in the block are present in the filter. This works as there's no such thing as a false negative (query says it isn't there and it's actually there) with this filter type still.

Reply:
Ah, I wasn't aware of this function. But it definitely makes sense to consolidate, which I've done now.